examples/fishjam-chat/screens/ConnectWithVideoRoomScreen.tsx

```diff
@@ -0,0 +1,186 @@
+import type { BottomTabScreenProps } from '@react-navigation/bottom-tabs';
+import { CompositeScreenProps } from '@react-navigation/native';
+import { NativeStackScreenProps } from '@react-navigation/native-stack';
+import React, { useEffect, useState } from 'react';
+import {
+  Dimensions,
+  Image,
+  KeyboardAvoidingView,
+  SafeAreaView,
+  StyleSheet,
+  Text,
+  View,
+} from 'react-native';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { Button, TextInput, DismissKeyboard } from '../components';
+import {
+  AppRootStackParamList,
+  TabParamList,
+} from '../navigators/AppNavigator';
+import { joinRoomWithRoomManager } from '../utils/roomManager';
+
+type Props = CompositeScreenProps<
+  BottomTabScreenProps<TabParamList, 'ConnectWithVideoRoom'>,
+  NativeStackScreenProps<AppRootStackParamList>
+>;
+
+type VideoRoomEnv = 'staging' | 'prod';
+
+type VideoRoomData = {
+  env: VideoRoomEnv;
+  roomName: string;
+  userName: string;
+};
+
+async function saveStorageData(videoRoomData: VideoRoomData) {
+  await AsyncStorage.setItem('videoRoomData', JSON.stringify(videoRoomData));
+}
+
+async function readStorageData(): Promise<VideoRoomData> {
+  const storageData = await AsyncStorage.getItem('videoRoomData');
+  if (storageData) {
+    const videoRoomData = JSON.parse(storageData) as VideoRoomData;
+    return videoRoomData;
+  }
+  return { env: 'staging', roomName: '', userName: '' };
+}
+
+export function shouldShowVideoRoomTab() {
+  return (
+    !!process.env.EXPO_PUBLIC_VIDEOROOM_STAGING_ROOM_MANAGER &&
+    !!process.env.EXPO_PUBLIC_VIDEOROOM_PRODUCTION_ROOM_MANAGER
+  );
+}
+
+/**
+ * Connect with the VideoRoom - our example service for video conferences
+ */
+export default function ConnectScreen({ navigation }: Props) {
+  const [connectionError, setConnectionError] = useState<string | null>(null);
+  const [env, setEnv] = useState<VideoRoomEnv>('staging');
+  const [loading, setLoading] = useState(false);
+
+  const [roomName, setRoomName] = useState('');
+  const [userName, setUserName] = useState('');
+
+  useEffect(() => {
+    async function readData() {
+      const {
+        env: storedEnv,
+        roomName: storedRoomName,
+        userName: storedUserName,
+      } = await readStorageData();
+
+      setRoomName(storedRoomName);
+      setUserName(storedUserName);
+      setEnv(storedEnv);
+    }
+    readData();
+  }, []);
+
+  const onTapConnectButton = async () => {
+    try {
+      setConnectionError(null);
+      setLoading(true);
+      const roomManagerUrl =
+        env === 'staging'
+          ? process.env.EXPO_PUBLIC_VIDEOROOM_STAGING_ROOM_MANAGER!
+          : process.env.EXPO_PUBLIC_VIDEOROOM_PRODUCTION_ROOM_MANAGER!;
+      saveStorageData({ env, roomName, userName });
```

Comment:
**question:** What is the purpose of the example app saving those details? Isn't it supposed to be as simple as possible?

Reply:
I also have problem with this. I would like to keep this app as simple as possible. But also for our internal use, it would be useful to have some app to test VIdeoRoom. So either we can create separate app for that. Or we can reuse example. And if someone would like to use this app for real VideoRoom - storing data will be super useful.

Another idea I had, was to move files around in project, to hide code related to VideoRoom (but to still keep it here). I'm open for any suggestions

Reply:
The only thing I don't understand is why we need to store `roomName` and `userName`. Can't this be hard coded when we use it? Like we do with other screens?

Reply:
What values would like to hardcode here? It is specific to user, I guess (your name and recently used room)